docs/tutorial-setup-dev.md

```diff
@@ -0,0 +1,464 @@
+
+For a quick reference, just jump to the [Development Setup](dev-setup) page, otherwise keep reading for more details.
+
+## Which tools should I use?
+
+As mentioned, you'll need a combination of both the Demisto UI and other tools. 
+
+As a general rule of the thumb, we recommend that you use an external IDE and toolchain when:
+- Working on your [integration code](code-conventions) (YourIntegration.py)
+- Working on the [unit test script](unit-testing) (YourIntegration_test.py)
+- Working on the [CHANGELOG.md](changelog) and README.md documentation files
+- Running the [linting](linting) and testing
+
+Instead, you should use the Demisto UI when:
+- Creating the [Test Playbooks](testing)
+- Auto-generate the [integration documentation](integration-docs)
+- Creating [example playbooks](playbooks)
+- Working on the properties of your integration (parameters, commands, arguments, outputs, etc.)
+- Testing the User Experience
+
+## What IDE should I use?
+
+When it comes to an External IDE, you should stick to what you're comfortable with.
+
+We developed a free [plugin](https://plugins.jetbrains.com/plugin/12093-demisto-add-on-for-pycharm) for [PyCharm](https://www.jetbrains.com/pycharm/) that simplifies/automates a few tasks such as:
+- Running unit tests
+- Creating a blank integration or automation script
+- Uploading/Downloading your integration code to/from Demisto
+- Running commands directly on Demisto
+
+However, if you want to a different IDE (Visual Studio Code, Sublime, vi, emacs, etc.) it's totally fine! It just means that some of those tasks must be performed manually. To automate them, you can use the  [demisto-sdk](https://github.com/demisto/demisto-sdk). In this tutorial, we will be using it for unit tests, but more features will come in the future.
+
+## Requirements
+
+Here are few requirements to make sure that you an easily build an Demisto Integration without running into issues down the road.
+
+### Demisto
+
+You need an instance of Demisto up and running. You can Sign Up for the [Demisto Free Edition](https://start.paloaltonetworks.com/sign-up-for-demisto-free-edition) or, if you're entitled to, contact your Business Development representative to have a non-production license.
+
+### Operating System
+
+So far we've been using the following Operating Systems:
+- MacOS
+- Linux
+- Windows (only with WSL - [Windows Subsystem For Linux](https://docs.microsoft.com/en-us/windows/wsl/about))
+
+If you successfully manage to get this work on other platforms (native Windows, OpenBSD, etc.) , please let us know and we'll add it to the tutorial! (click on Report an issue at the bottom of this page).
+
+### Python
+
+You will need to build your integration using **Python** and, more specifically, Python 3.6+. While some content is built via Javascript and Python 2, we require Python 3.6+ for contributions.
```

Comment:
```suggestion
You will need to build your integration using **Python** and, more specifically, Python 3.7+. While some content is built via Javascript and Python 2, we require Python 3.7+ for contributions.
```

Let's go with python 3.7. I think we started using some features of 3.7 in some our scripts.